### PR TITLE
updateURL: Allow overriding for CI builds

### PR DESCRIPTION
**Pull request by celebdor**
# Description

    In order to be able to mirror 4.next builds and the latest nightly
    releases, it is handy to have an environment variable for CI/devs to
    mirror from development registry graphs.
    
    The two env variables that this defines are:
    UPDATE_URL_OVERRIDE: URL to a release graph
    VERSION_FILTER: A golang regular expression to filter out disconnected
    graphs that are provided by the aforementioned URL.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally:
UPDATE_URL_OVERRIDE=https://{redacted}/graph oc-mirror -c imageset.yaml --dest-skip-tls docker://${HOSTNAME}:5000/

imageset:
```
kind: ImageSetConfiguration
apiVersion: mirror.openshift.io/v1alpha2
archiveSize: 4
storageConfig:
  registry:
    imageURL: toni.openshift-agent.cloud:5000/mirror/cicandidate-oc-mirror-metadata
    skipTLS: true
mirror:
  platform:
    architectures:
      - "amd64"
    channels:
      - name: stable-4.12
        type: ocp
  additionalImages:
    - name: registry.redhat.io/ubi8/ubi:latest
```

sample image retrieved:
sha256:323e7d72b8bd127ece7ef25e602ab924977f37c4d3516cf327b7ab6b1b88b669 toni.openshift-agent.cloud:5000/openshift/release:4.12.0-ec.3-x86_64-installer       

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jpower432 — APPROVED** [review]
Nice add @celebdor

**Review by soltysh — CHANGES_REQUESTED** [review]
The change looks reasonable, but my main question is do we want to expose that functionality in a supported way to our users, if not then every invocation should throw a warning that this is not a supported operation, such that we can safely ignore it in CI, but users will be warned that it's not something we will fix for them if it doesn't work they way they expect.

/lgtm cancel

File: pkg/cincinnati/cincinnati.go [hunk 1]
```diff
@@ -337,9 +339,24 @@ func GetChannelMinOrMax(ctx context.Context, c Client, arch string, channel stri
 	}
 
 	// Find the all versions within the graph.
+	var versionMatcher *regexp.Regexp
+	if versionFilter := os.Getenv("VERSION_FILTER"); len(versionFilter) != 0 {
+		klog.Info("Usage of the VERSION_FILTER environment variable is unsupported")
```
Comment (soltysh) [inline pkg/cincinnati/cincinnati.go:344]: Nit, worth using `klog.Warning` for it next time.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**